### PR TITLE
XIP-36: Frames Support in Review status

### PR DESCRIPTION
**Pull request by jhaaaa**
Ready for Review status.

Left a few comments for review by @neekolas and @daria-github - thank you!

### CODE REVIEW

File: XIPs/xip-36-frames-support.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+At a high level, what developers are doing with all this code is validating three things:
+
+1. The incoming POST was from a legitimate button click
+2. Which button was clicked
+3. Who clicked the button
+
+With this information, the developer then responds with HTML outlining an update to the Frame’s state using the same format as the initial render.
+
+In Jan 2024, Coinbase [released `FrameKit`](https://github.com/coinbase/onchainkit), with some high-level APIs designed to verify
+Frame POST payloads. Those APIs are `getFrameAccountAddress`, `getFrameMetadata`, and `getFrameValidatedMessage`. XMTP already has all the primitives needed to craft payloads that can enable these same workflows.
```
Comment (jhaaaa) [inline XIPs/xip-36-frames-support.md:161]:
```suggestion
In Jan 2024, Coinbase [released `FrameKit`](https://github.com/coinbase/onchainkit), with some high-level APIs designed to verify Frame POST payloads. Those APIs are `getFrameAccountAddress`, `getFrameMetadata`, and `getFrameValidatedMessage`. XMTP already has all the primitives needed to craft payloads that can enable these same workflows.
```

File: XIPs/xip-36-frames-support.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+1. [Rendering/interaction helper library](https://github.com/xmtp/xmtp-web/tree/main/packages/frames-client)  
+Needs some updates to accommodate changes to the POST message schema made after development began
+
+2. [POST payload verification helper library](https://github.com/xmtp/xmtp-node-js-tools/tree/main/packages/frames-validator)  
+Needs some updates to accommodate changes to the POST message schema made after development began
+
+3. [Open Graph Proxy service](https://github.com/neekolas/og-proxy)  
+Still missing support for proxying requests for images. This support will be a hard requirement for launch.
+
+Here are some specifications you might also want to explore as a part of working with these reference implementations:
```
Comment (jhaaaa) [inline XIPs/xip-36-frames-support.md:235]: See below - this is where I added the link to the Open Frames spec as suggested by Nick. 

I also added a link to the Farcaster Frames spec, just to be comprehensive.

File: XIPs/xip-36-frames-support.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+The Frames spec for creating interactive and authenticated experiences on Farcaster, embeddable in any Farcaster client
+
+- [Open Frames specification](https://github.com/open-frames/standard)  
+A lightweight extension to the Frames spec to help enable non-Farcaster apps and protocols to support Frames.
+
+## Security considerations
+
+In the Farcaster model of Frames, messages are signed on a server by the app and the server makes the POST request to the Frame URL. This means the Frame app developer only sees requests from the server IP and never the client.
+
+However, Farcaster clients, like Warpcast, download the Frame image directly. This [leaks the IP address of the viewer to the Frame developer](https://twitter.com/danfinlay/status/1752500815200399633?t=1psurRGh-JTii3AHjvUrMg). According to a reply on X, they are "[working on it](https://x.com/dwr/status/1752506807548072140?s=20)."
```
Comment (jhaaaa) [inline XIPs/xip-36-frames-support.md:247]: Do we know if they've fixed this?

File: XIPs/xip-36-frames-support.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+- All Frames will be sent using ordinary XMTP text messages.
+- Client apps can choose to expand URLs found in these messages into Frames at their discretion.
+- No new Content Types are required.
+- No new SDK changes are necessary for XMTP SDKs that allow app developers to sign messages with the XMTP identity key.
+- The new helper libraries that facilitate Frame rendering, interaction, and verification are strictly opt-in.
+
+## Reference implementations
+
+We've developed reference implementations for all three components required to enable Frames on XMTP:
```
Comment (jhaaaa) [inline XIPs/xip-36-frames-support.md:224]: For each of the 3 items below, we call out areas that need updates. Might we have any updates to share?

**Review by neekolas — APPROVED** [review]

**Pull request merged**